packages/serverpod_client/lib/src/serverpod_client_shared.dart

```diff
@@ -427,6 +429,54 @@ abstract class ServerpodClientShared extends EndpointCaller {
     await _sendControlCommandToStream('auth', {'key': authKey});
   }
 
+  /// Performs the actual call to the server.
+  /// Subclasses implement bespoke logic for e.g. web / io clients.
+  Future<String> callServerEndpointImpl<T>(
```

Comment:
Suggestion: Make this method private.
This is only an implementation detail that should be called internally. We don't need to expose it in the interface.

Reply:
Refactored according to conversation, replacing inheritance with composition.